front-end-tools/src/components/DeployComponent.tsx

```diff
@@ -0,0 +1,268 @@
+                        setIsModuleReferenceAlreadyDeployedStep1(false);
+                    } else {
+                        setIsModuleReferenceAlreadyDeployedStep1(true);
+                    }
+                })
+                .catch(() => {
+                    setIsModuleReferenceAlreadyDeployedStep1(false);
+                });
+        }
+    }, [connection, account, client, moduleReferenceCalculated]);
+
+    function onSubmit() {
+        setTxHashDeploy(undefined);
+        setTransactionErrorDeploy(undefined);
+
+        // Send deploy transaction
+
+        const tx = deploy(connection, account, base64Module);
+        tx.then((txHash) => {
+            setModuleReferenceDeployed(undefined);
+            setTxHashDeploy(txHash);
+        }).catch((err: Error) => setTransactionErrorDeploy((err as Error).message));
+    }
+
+    return (
+        <Box header="Step 1: Deploy Smart Contract Module">
+            <Form onSubmit={form.handleSubmit(onSubmit)}>
+                <Form.Group className="mb-3">
+                    <Form.Label>Upload Smart Contract Module File (e.g. myContract.wasm.v1)</Form.Label>
+                    <Form.Control
+                        type="file"
+                        accept=".wasm,.wasm.v0,.wasm.v1"
+                        {...form.register('file')}
+                        onChange={async (e) => {
+                            const register = form.register('file');
+
+                            register.onChange(e);
+
+                            setUploadError(undefined);
+                            setModuleReferenceDeployed(undefined);
+                            setTransactionErrorDeploy(undefined);
+                            setTxHashDeploy(undefined);
+
+                            const files = form.getValues('file');
+
+                            if (files !== undefined && files !== null && files.length > 0) {
+                                const file = files[0];
+                                const arrayBuffer = await file.arrayBuffer();
+
+                                const module = btoa(
+                                    new Uint8Array(arrayBuffer).reduce((data, byte) => {
+                                        return data + String.fromCharCode(byte);
+                                    }, '')
+                                );
+
+                                setBase64Module(module);
+                                setModuleReferenceCalculated(
+                                    Buffer.from(sha256([new Uint8Array(arrayBuffer)])).toString('hex')
+                                );
+
+                                // Concordium's tooling create versioned modules e.g. `.wasm.v1` now.
+                                // Unversioned modules `.wasm` cannot be created by Concordium's tooling anymore.
+                                // If the module is versioned, the first 8 bytes are the version, followed by the `magicValue` below.
```

Comment:
The first 4 bytes are the version I think, the next 4 are the length, followed by the Wasm module.